extension/core_functions/scalar/list/array_slice.cpp

```diff
@@ -59,18 +59,25 @@ static idx_t CalculateSliceLength(idx_t begin, idx_t end, INDEX_TYPE step, bool
 }
 
 template <typename INPUT_TYPE, typename INDEX_TYPE>
-INDEX_TYPE ValueLength(const INPUT_TYPE &value) {
+INDEX_TYPE ValueLength(const INPUT_TYPE &value, const LogicalTypeId &) {
```

Comment:
Indeed, I also don't see a way around passing the logical type.
Minor nit: Should we directly pass the `LogicalTypeId` by value (`uint8_t`)? To save some bytes...